cpp/src/io/utilities/hostdevice_vector.hpp

```diff
@@ -34,6 +34,11 @@ class hostdevice_vector {
  public:
   using value_type = T;
 
+  hostdevice_vector() {}
+
+  hostdevice_vector(hostdevice_vector &&v) { move(std::move(v)); }
+  void operator=(hostdevice_vector &&v) { move(std::move(v)); }
```

Comment:
Is there a reason why this does not return a reference?
```suggestion
  hostdevice_vector& operator=(hostdevice_vector &&v) { move(std::move(v)); return *this; }
```